pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -3929,16 +3926,106 @@ func TestChangefeedEnrichedAvro(t *testing.T) {
 
 		assertionKey := `{"a":{"long":0}}`
 		assertionAfter := `"after": {"foo": {"a": {"long": 0}, "b": {"string": "dog"}}}`
-		assertionSource := fmt.Sprintf(`"source": {"source": {"job_id": {"string": "%d"}}}`, jobID)
 
-		assertPayloadsEnvelopeStripTs(t, foo, changefeedbase.OptEnvelopeEnriched, []string{
-			fmt.Sprintf(`foo: %s->{%s, "op": {"string": "c"}, %s}`,
-				assertionKey, assertionAfter, assertionSource),
-		})
+		assertPayloadsEnriched(t, foo, []string{
+			fmt.Sprintf(`foo: %s->{%s, "op": {"string": "c"}}`,
+				assertionKey, assertionAfter),
+		}, nil)
 	}
 	cdcTest(t, testFn, feedTestForceSink("kafka"))
 }
 
+func TestChangefeedEnrichedSourceWithNodeAndClusterInfo(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	cases := []struct {
+		name            string
+		format          string
+		expectedMessage string
+		supportedSinks  []string
+	}{
+		{name: "json", format: "json", expectedMessage: `foo: {"i": 0}->{"after": {"i": 0}, "op": "c"}`, supportedSinks: []string{"kafka", "pubsub", "sinkless"}},
+		// TODO(#139660): the webhook sink forces topic_in_value, but
+		// this is not supported by the enriched envelope type. We should adapt
+		// the test framework to account for this.
+		{name: "json-webhook", format: "json", expectedMessage: `: {"i": 0}->{"after": {"i": 0}, "op": "c"}`, supportedSinks: []string{"webhook"}},
+		{name: "avro", format: "avro", expectedMessage: `foo: {"i":{"long":0}}->{"after": {"foo": {"i": {"long": 0}}}, "op": {"string": "c"}}`, supportedSinks: []string{"kafka"}},
+	}
+
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			clusterName := "clusterName123"
+			testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+				clusterID := s.Server.ExecutorConfig().(sql.ExecutorConfig).NodeInfo.LogicalClusterID().String()
+
+				// Test clusters will be given the latest db version by default.
+				// Test clusters will also run versions which have had the dev offset
+				// applied. E.g. this version can start 1000025.1 as opposed to 25.1.
+				dbVersion := clusterversion.Latest.String()
+				sqlDB := sqlutils.MakeSQLRunner(s.DB)
+
+				sqlDB.Exec(t, `CREATE TABLE foo (i INT PRIMARY KEY)`)
+				sqlDB.Exec(t, `INSERT INTO foo values (0)`)
+				testFeed := feed(t, f, fmt.Sprintf(`CREATE CHANGEFEED FOR foo WITH envelope=enriched, enriched_properties='source', format=%s`, testCase.format))
+				defer closeFeed(t, testFeed)
+
+				var jobID int64
+				var nodeName string
+				var sourceAssertion func(actualSource map[string]interface{})
+				if ef, ok := testFeed.(cdctest.EnterpriseTestFeed); ok {
+					jobID = int64(ef.JobID())
+				}
+				sqlDB.QueryRow(t, `SELECT value FROM crdb_internal.node_runtime_info where component = 'DB' and field = 'Host'`).Scan(&nodeName)
+
+				if testCase.format == "json" {
```

Comment:
nit: you can move this `if` condition inside `sourceAssertion` to save some lines of code